tests/e2e/performance/test_small_file_workload.py

```diff
@@ -500,43 +424,33 @@ def test_smallfile_workload(
+                "storageclass": self.crd_data["spec"]["workload"]["args"][
+                    "storageclass"
+                ],
+                "vol_size": self.crd_data["spec"]["workload"]["args"]["storagesize"],
             },
         )
 
-        while True:
-            logs = bench_pod.exec_oc_cmd(
-                f"logs {small_file_client_pod}", out_yaml_format=False
-            )
-            if "RUN STATUS DONE" in logs:
-                # Getting the end time of the test
-                end_time = time.strftime("%Y-%m-%dT%H:%M:%SGMT", time.gmtime())
-                full_results.add_key(
-                    "test_time", {"start": start_time, "end": end_time}
-                )
-                # if Internal ES is exists, Copy all data from the Internal to main ES
-                if es:
-                    log.info("Copy all data from Internal ES to Main ES")
-                    es._copy(full_results.es)
-                full_results.read()
-                if not full_results.dont_check:
-                    full_results.add_key("hosts", full_results.get_clients_list())
-                    full_results.init_full_results()
-                    full_results.aggregate_host_results()
-                    test_status = full_results.aggregate_samples_results()
-                    full_results.es_write()
-
-                    # Creating full link to the results on the ES server
-                    log.info(
-                        f"The Result can be found at ; {full_results.results_link()}"
-                    )
-                else:
-                    test_status = True
+        self.wait_for_wl_to_finish(timeout=3600, sleep=30)
 
-                break
-
-            if timeout < (time.time() - test_start_time):
-                raise TimeoutError("Timed out waiting for benchmark to complete")
-            time.sleep(30)
-        assert not get_logs_with_errors() and test_status, "Test Failed"
+        if "RUN STATUS DONE" in self.test_logs:
+            # Getting the end time of the test
+            full_results.add_key(
+                "test_time", {"start": self.start_time, "end": self.end_time}
+            )
+            # if Internal ES is exists, Copy all data from the Internal to main ES
+            self.copy_es_data(es)
+            full_results.read()
+            if not full_results.dont_check:
+                full_results.add_key("hosts", full_results.get_clients_list())
+                full_results.init_full_results()
+                full_results.aggregate_host_results()
+                test_status = full_results.aggregate_samples_results()
+                full_results.es_write()
+
+                # Creating full link to the results on the ES server
+                log.info(f"The Result can be found at : {full_results.results_link()}")
+            else:
+                test_status = True
+
+        assert test_status, "Test Failed !"
```

Comment:
is this indentation correct?

Reply:
yes it is, the assert is in the end, outside the if statement which start on line 436